slsim/image_simulation.py

```diff
@@ -395,10 +417,180 @@ def point_source_image_with_variability(
             mag_zero_point=mag_zero,
             delta_pix=delta_pix,
             num_pix=num_pix,
-            psf_kernels=psf_kernel,
+            psf_kernel=psf_kernel,
             transform_pix2angle=transf_matrix,
             time=time,
         )
         all_image.append(image_test)
     variab_images = [list(x) for x in zip(*all_image)]
-    return variab_images
+    image = np.array(variab_images)
+    n = len(image)
+    subarray_shape = image[0].shape
+    results = np.zeros(subarray_shape)
+    for i in range(n):
+        results += image[i]
+    return results
+
+
+def deflector_images_with_different_zeropoint(
+    lens_class, band, mag_zero_point, delta_pix, num_pix
+):
+    """Creates deflector images with different magnitude zero point.
+
+    :param lens_class: Lens() object
+    :param band: imaging band
+    :param mag_zero_point: magnitude zero point in band
+    :param delta_pix: pixel scale of image generated
+    :param num_pix: number of pixels per axis
+    :returns: list of deflector images with different zero point
+    """
+    image = []
+    for mag_zero in mag_zero_point:
+        image.append(
+            sharp_image(
+                lens_class=lens_class,
+                band=band,
+                mag_zero_point=mag_zero,
+                delta_pix=delta_pix,
+                num_pix=num_pix,
+            )
+        )
+    return image
+
+
+def image_plus_poisson_noise(image, exposure_time):
+    """Creates an image with possion noise.
+
+    :param image: image or list of image
+    :param exposure_time: exposure time for each image or exposure map
+    :return: image with possion noise
+    """
+    if isinstance(image, list):
+        expo_time = exposure_time
+        image_plus_noise = []
+        for i in range(len(image)):
+            data = image[i]
```

Comment:
you have twice basically the same code. Avoid duplication and have a definition for a single band and then another one that loops through them. Currently it is ambiguous with different input and output formats